backend-api/src/functions/asyncActiveSession/tokenService/tests/tokenService.test.ts

```diff
@@ -1,443 +1,334 @@
-  MockPubicKeyGetterGetPublicKeySuccess,
-  MockTokenVerifierVerifyError,
-  MockTokenVerifierVerifySuccess,
-} from "./mocks";
+import { MockJWTBuilder } from "../../../testUtils/mockJwtBuilder";
+import { ExpectedClaims, ITokenService, TokenService } from "../tokenService";
+import { MockTokenVerifierError, MockTokenVerifierSuccess } from "./mocks";
 
 describe("Token Service", () => {
-  let mockFetch: jest.SpyInstance;
   let tokenService: ITokenService;
-  let dependencies: ITokenServiceDependencies;
+  let mockExpectedClaims: ExpectedClaims;
 
-  beforeEach(() => {
-    dependencies = {
-      publicKeyGetter: () => new MockPubicKeyGetterGetPublicKeySuccess(),
-      tokenVerifier: () => new MockTokenVerifierVerifySuccess(),
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date("2024-03-10"));
+  });
+
+  beforeEach(async () => {
+    tokenService = new TokenService("testJwksUri", {
+      tokenVerifier: new MockTokenVerifierSuccess(),
+    });
+    mockExpectedClaims = {
+      aud: "mockAudience",
+      iss: "mockIssuer",
+      scope: "idCheck.activeSession.read",
     };
-    tokenService = new TokenService(dependencies);
-    mockFetch = jest.spyOn(global, "fetch").mockImplementation(() =>
-      Promise.resolve({
-        status: 200,
-        ok: true,
-        headers: new Headers({
-          header: "mockHeader",
-        }),
-        text: () =>
-          Promise.resolve(
-            JSON.stringify({
-              keys: [
-                {
-                  kty: "mockKty",
-                  x: "mockX",
-                  y: "mockY",
-                  crv: "mockCrv",
-                  d: "mockD",
-                  kid: "mockKid",
-                },
-              ],
-            }),
-          ),
-      } as Response),
-    );
   });
 
-  afterEach(() => {
-    mockFetch.mockRestore();
+  describe("Header", () => {
+    describe("Given the JWT header cannot be decoded", () => {
+      it("Returns a client error result", async () => {
+        const result = await tokenService.validateServiceToken(
+          "test.invalid.jwt",
+          mockExpectedClaims,
+        );
+
+        expect(result.isError).toBe(true);
+        expect(result.value).toStrictEqual({
+          errorMessage: "Failed to decode token header",
+          errorCategory: "CLIENT_ERROR",
+        });
+      });
+    });
+
+    describe("Given the kid claim is not present in JWT header", () => {
+      it("Returns a client error result", async () => {
+        const mockEncodedJwt = new MockJWTBuilder().deleteKid().getEncodedJwt();
+        const result = await tokenService.validateServiceToken(
+          mockEncodedJwt,
+          mockExpectedClaims,
+        );
+
+        expect(result.isError).toBe(true);
+        expect(result.value).toStrictEqual({
+          errorMessage: "Invalid kid claim",
+          errorCategory: "CLIENT_ERROR",
+        });
+      });
+    });
   });
 
-  describe("Get Sub From Token", () => {
```

Comment:
These tests have moved - they are no longer required in the TokenService. They are now in the sendHttpRequest and the publicKeyGetter test files...